File: java/src/main/java/com/authdog/resources/FeatureFlagsResource.java
package com.authdog.resources;

import com.authdog.AuthdogClient;
import com.authdog.exceptions.ApiException;
import com.authdog.exceptions.AuthenticationException;
import com.fasterxml.jackson.databind.JsonNode;

/**
 * Environment feature flag operations.
 */
public final class FeatureFlagsResource {
    /**
     * Authdog client.
     */
    private final AuthdogClient client;

    /**
     * Create a feature flags helper.
     * @param clientParam Authdog client
     */
    public FeatureFlagsResource(final AuthdogClient clientParam) {
        this.client = clientParam;
    }

    /**
     * List feature flags.
     * @param tenantId tenant ID
     * @param environmentId environment ID
     * @return flags envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode list(final String tenantId,
                         final String environmentId)
            throws AuthenticationException, ApiException {
        return client.request("GET",
                EnvPaths.prefix(tenantId, environmentId)
                        + "/feature-flags",
                null, null, JsonNode.class);
    }

    /**
     * Create or update a feature flag.
     * @param tenantId tenant ID
     * @param environmentId environment ID
     * @param body request body
     * @return flag envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode save(final String tenantId,
                         final String environmentId,
                         final Object body)
            throws AuthenticationException, ApiException {
        return client.request("POST",
                EnvPaths.prefix(tenantId, environmentId)
                        + "/feature-flags",
                body, null, JsonNode.class);
    }

    /**
     * Delete a feature flag.
     * @param tenantId tenant ID
     * @param environmentId environment ID
     * @param flagId flag ID
     * @return delete envelope
     * @throws AuthenticationException when unauthorized
     * @throws ApiException when the request fails
     */
    public JsonNode delete(final String tenantId,
                           final String environmentId,
                           final String flagId)
            throws AuthenticationException, ApiException {
        return client.request("DELETE",
                EnvPaths.prefix(tenantId, environmentId)
                        + "/feature-flags/" + flagId,
                null, null, JsonNode.class);
    }
}

File: csharp/Resources/FeatureFlagsResource.cs
using System.Net.Http;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;

namespace Authdog
{
    /// <summary>
    /// Environment feature-flag Wave 3 operations
    /// </summary>
    public class FeatureFlagsResource
    {
        private readonly AuthdogClient _client;

        public FeatureFlagsResource(AuthdogClient client)
        {
            _client = client;
        }

        public Task<JObject> ListAsync(string tenantId, string environmentId) =>
            _client.RequestAsync<JObject>(
                HttpMethod.Get,
                $"{AuthdogClient.Env(tenantId, environmentId)}/feature-flags");

        public JObject List(string tenantId, string environmentId) =>
            ListAsync(tenantId, environmentId).GetAwaiter().GetResult();

        public Task<JObject> SaveAsync(string tenantId, string environmentId, object body) =>
            _client.RequestAsync<JObject>(
                HttpMethod.Post,
                $"{AuthdogClient.Env(tenantId, environmentId)}/feature-flags",
                body);

        public JObject Save(string tenantId, string environmentId, object body) =>
            SaveAsync(tenantId, environmentId, body).GetAwaiter().GetResult();

        public Task<JObject> DeleteAsync(string tenantId, string environmentId, string flagId) =>
            _client.RequestAsync<JObject>(
                HttpMethod.Delete,
                $"{AuthdogClient.Env(tenantId, environmentId)}/feature-flags/{flagId}");

        public JObject Delete(string tenantId, string environmentId, string flagId) =>
            DeleteAsync(tenantId, environmentId, flagId).GetAwaiter().GetResult();
    }
}
